gyronimo/metrics/morphism_cylindrical.cc

```diff
@@ -0,0 +1,122 @@
+// ::gyronimo:: - gyromotion for the people, by the people -
+// An object-oriented library for gyromotion applications in plasma physics.
+// Copyright (C) 2022 Paulo Rodrigues.
+
+// ::gyronimo:: is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// ::gyronimo:: is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with ::gyronimo::.  If not, see <https://www.gnu.org/licenses/>.
+
+// @morphism_cylindrical.cc, this file is part of ::gyronimo::
+
+#include <gyronimo/metrics/morphism_cylindrical.hh>
+
+namespace gyronimo {
+
+//! Maps cylindrical coordinates @f$ q^\alpha @f$ into cartesian coordinates @f$ \textbf{x} @f$.
+/*!
+	In cylindrical coordinates, @f$ \left( q^1, q^2, q^3 \right) = \left( r, \phi, z \right) @f$.
+	Implements the coordinate transformation:
+	@f{gather*}{
+		\begin{aligned}
+			x &= r \, \cos \phi \\
+			y &= r \, \sin \phi \\
+			z &= z
+		\end{aligned}
+    @f}
+*/
+IR3 morphism_cylindrical::operator()(const IR3 &q) const {
+	return {q[IR3::u] * std::cos(q[IR3::v]), q[IR3::u] * std::sin(q[IR3::v]), q[IR3::w]};
+}
+
+//! Inverse transform from cartesian coordinates @f$ \textbf{x} @f$ into cylindrical coordinates @f$ q^\alpha @f$.
+/*!
+	Implements the inverse transformation:
+	@f{gather*}{
+		\begin{aligned}
+			r &= \sqrt{x^2 + y^2} \\
+			\phi &= \arctan \left( \frac{y}{x} \right) \\
+			z &= z
+		\end{aligned}
+    @f}
+*/
+IR3 morphism_cylindrical::inverse(const IR3 &x) const {
+	return {sqrt(x[IR3::u]*x[IR3::u] + x[IR3::v]*x[IR3::v]),
+			atan2(x[IR3::v], x[IR3::u]), x[IR3::w]};
```

Comment:
You should need a std:: qualifier here... Am I missing a using namespace std (which should be avoided, if possible) somewhere?

Reply:
I did not write `using namespace std; ` anywhere. I don't think `std::sqrt` needs the qualifier, but I will add it anyway for the sake of consistency and to make sure it is using the right one.